include/bytestring-cpp-macros.h

```diff
@@ -7,7 +7,8 @@
     || ((defined(arm_HOST_ARCH) || defined(aarch64_HOST_ARCH)) \
         && defined(__ARM_FEATURE_UNALIGNED)) \
     || defined(powerpc_HOST_ARCH) || defined(powerpc64_HOST_ARCH) \
-    || defined(powerpc64le_HOST_ARCH)
+    || defined(powerpc64le_HOST_ARCH) \
+    || defined(PURE_HASKELL)
```

Comment:
I understand that due to the unaligned static tables, we do not care to support building with the pure-haskell flag on hosts that are not unaligned-friendly.

But the right way to say that is to write an `#error` pragma guarded by `PURE_HASKELL` in the `#else` branch reached on hosts not known to be unaligned-friendly.

(And then you should probably add javascript as a known unaligned-friendly host.)

Reply:
Actually I don't remember why just adding javascript to the list isn't enough. Perhaps CI will remind me.

Reply:
It's not enough because if someone tries to build with `PURE_HASKELL` for an unaligned-unfriendly host,
* There will be no fallback implementations for the stuff in `Data.ByteString.Utils.UnalignedWrite`.
* The buffers `digit_pairs_table` and `lower_hex_table` are not guaranteed to actually be aligned when building with `PURE_HASKELL`, but we still perform aligned read ops on them.